kratos/elements/distance_calculation_element_simplex.h

```diff
@@ -212,13 +206,15 @@ class DistanceCalculationElementSimplex : public Element
 //			if (distances[i] >= 0.0 && distances[i] < 1e-30) distances[i] = 1e-30;
         }
 
+       const double dgauss = inner_prod(N,distances);
+
         const unsigned int step = rCurrentProcessInfo[FRACTIONAL_STEP];
     
         if(step == 1) //solve a poisson problem with a positive/negative heat source depending on the sign of the existing distance function
         {
             //compute distance on gauss point
-            const double dgauss = inner_prod(N,distances);
-            
+
+            this->GetValue(DISTANCE) = dgauss; //saving the distance, to see if it changed sign between iterations
```

Comment:
@pablobecker two comments here:

1 - if you want to do this, you should do

this->SetValue(DISTANCE,dgauss)

otherwise this will be an error when running in parallel in FULL_DEBUG

2 - this is only needed for debugging purposes right?